common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -4086,6 +4086,13 @@
     ],
     "sqlState" : "42K0L"
   },
+  "LATERAL_JOIN_WITH_ARROW_UDTF_UNSUPPORTED" : {
+    "message" : [
+      "LATERAL JOIN with Arrow-optimized user-defined table functions (UDTFs) is not supported. Arrow UDTFs cannot be used on the right-hand side of a lateral join.",
```

Comment:
why arrow UDTF cannot support lateral join?
is it because the row id is missing? Just wondering can we provide a arrow column containing the row id, so that the JVM side can join input rows and output rows again

Reply:
Yes that's one option to support lateral join with arrow UDTFs, but I expect most of the use cases will be via TABLE arguments.